components/integrations/abstract_command.tsx

```diff
@@ -14,79 +14,63 @@ import SpinnerButton from 'components/spinner_button';
-        * The spinner loading text to render, has id and defaultMessage
-        */
-        loading: PropTypes.object.isRequired,
-
-        /**
-         * Any extra component/node to render
-         */
-        renderExtra: PropTypes.node.isRequired,
-
-        /**
-        * The server error text after a failed action
-        */
-        serverError: PropTypes.string.isRequired,
-
-        /**
-        * The Command used to set the initial state
-        */
-        initialCommand: PropTypes.object,
-
-        /**
-        * The async function to run when the action button is pressed
-        */
-        action: PropTypes.func.isRequired,
-    }
+type Props = {
+    team: Team;
+    header: {id: string ; defaultMessage: string};
+    footer: {id: string ; defaultMessage: string};
+    loading: {id: string ; defaultMessage: string};
+    renderExtra?: JSX.Element;
+    serverError: string;
+    initialCommand?: Partial<Command>;
+    action: (command: Command) => Promise<void>;
+}
 
-    constructor(props) {
+type State= {
+    saving: boolean;
+    clientError: null | JSX.Element | string;
+    trigger: string;
+    displayName: string;
+    description: string;
+    url: string;
+    method: 'P' | 'G' | '';
+    username: string;
+    iconUrl: string;
+    autocomplete: boolean;
+    autocompleteHint: string;
+    autocompleteDescription: string;
+}
+
+export default class AbstractCommand extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
         super(props);
 
         this.state = this.getStateFromCommand(this.props.initialCommand || {});
     }
 
-    getStateFromCommand = (command) => {
+    getStateFromCommand = (command: Props['initialCommand']) => {
         return {
-            displayName: command.display_name || '',
-            description: command.description || '',
-            trigger: command.trigger || '',
-            url: command.url || '',
-            method: command.method || REQUEST_POST,
-            username: command.username || '',
-            iconUrl: command.icon_url || '',
-            autocomplete: command.auto_complete || false,
-            autocompleteHint: command.auto_complete_hint || '',
-            autocompleteDescription: command.auto_complete_desc || '',
+            displayName: command?.display_name ?? '',
+            description: command?.description ?? '',
+            trigger: command?.trigger ?? '',
+            url: command?.url ?? '',
+            method: command?.method ?? REQUEST_POST,
+            username: command?.username ?? '',
+            iconUrl: command?.icon_url ?? '',
+            autocomplete: command?.auto_complete ?? false,
+            autocompleteHint: command?.auto_complete_hint ?? '',
+            autocompleteDescription: command?.auto_complete_desc ?? '',
             saving: false,
             clientError: null,
         };
     }
 
-    handleSubmit = (e) => {
-        e.preventDefault();
+    handleSubmit = (e?: React.MouseEvent) => {
```

Comment:
nit
```suggestion
    handleSubmit = (e?: MouseEvent) => {
```
where MouseEvent is imported from React

Reply:
(suggestion) Since we're using only `preventDefault` in the event, I'd go with an explicit interface here like `e: { preventDefault: () => void}`. This has the benefit that you won't need the casting to `any` in the tests you've written and the eslint ignore is not needed, too.

Finally, regarding the need for the event to be optional, it's stemming from the replacement of `onSubmit={this.handleSubmit}` with `onSubmit={() => this.handleSubmit}`. Could you please revert that change?

Also, it makes obvious which part of the vent interface we're using.